tests/ArrayBasedFormatterTest.php

```diff
@@ -398,4 +408,30 @@ public function array_of_null_serializes_cleanly_validate(mixed $serialized): vo
 
         self::assertNull($toTest['arr'][0]);
     }
+
+    public function nullable_property_set_properly_validate(mixed $serialized): void
+    {
+        $toTest = $this->arrayify($serialized);
+
+        self::assertArrayHasKey('str', $toTest);
+        self::assertArrayHasKey('int', $toTest);
+        self::assertArrayHasKey('float', $toTest);
+        self::assertArrayHasKey('bool', $toTest);
+        self::assertArrayHasKey('arr', $toTest);
+
+        self::assertNull($toTest['str']);
+        self::assertNull($toTest['int']);
+        self::assertNull($toTest['float']);
+        self::assertNull($toTest['bool']);
+        self::assertNull($toTest['arr']);
```

Comment:
I'm pretty sure `assertNull` is sufficient here.  If the key doesn't exist, trying to read from it in the `assertNull` block will fail with a warning anyway, which is still a useful fail.  That makes the `arrayHasKey` block unnecessary.